mlflow/server/auth/sqlalchemy_store.py

```diff
@@ -0,0 +1,142 @@
+    permission = Column(String(255))
+
+    def to_mlflow_entity(self):
+        return RegisteredModelPermission(
+            name=self.name,
+            user_id=self.user_id,
+            permission=self.permission,
+        )
+
+
+class SqlAlchemyStore:
+    db_uri = None
+    db_type = None
+    engine = None
+    ManagedSessionMaker = None
+
+    def init_db(self, db_uri):
+        self.db_uri = db_uri
+        self.db_type = extract_db_type_from_uri(db_uri)
+        self.engine = create_sqlalchemy_engine_with_retry(db_uri)
+        Base.metadata.create_all(bind=self.engine)
+        Base.metadata.bind = self.engine
+        SessionMaker = sessionmaker(bind=self.engine)
+        self.ManagedSessionMaker = _get_managed_session_maker(SessionMaker, self.db_type)
+
+    def authenticate_user(self, username: str, password: str) -> bool:
+        with self.ManagedSessionMaker() as session:
+            try:
+                user = self._get_user(session, username)
+                return check_password_hash(user.password_hash, password)
+            except MlflowException:
+                return False
+
+    def create_user(self, username: str, password: str, is_admin: bool = False):
+        _validate_username(username)
+        pwhash = generate_password_hash(password)
+        with self.ManagedSessionMaker() as session:
+            try:
+                user = SqlUser(username=username, password_hash=pwhash, is_admin=is_admin)
+                session.add(user)
+                session.flush()
+                return user.to_mlflow_entity()
+            except IntegrityError as e:
+                raise MlflowException(
+                    f"User (username={username}) already exists. Error: {str(e)}",
+                    RESOURCE_ALREADY_EXISTS,
+                )
+
+    @classmethod
+    def _get_user(cls, session, username: str) -> SqlUser:
+        try:
+            return session.query(SqlUser).filter(SqlUser.username == username).one()
+        except NoResultFound:
+            raise MlflowException(
+                f"User with username={username} not found",
+                RESOURCE_DOES_NOT_EXIST,
+            )
+        except MultipleResultsFound:
+            raise MlflowException(
+                f"Found multiple users with username={username}",
+                INVALID_STATE,
+            )
+
+    def has_user(self, username: str) -> bool:
+        with self.ManagedSessionMaker() as session:
+            return session.query(SqlUser).filter(SqlUser.username == username).first() is not None
+
+    def get_user(self, username: str) -> User:
+        with self.ManagedSessionMaker() as session:
+            return self._get_user(session, username).to_mlflow_entity()
+
+    def list_users(self) -> List[User]:
+        with self.ManagedSessionMaker() as session:
+            users = session.query(SqlUser).all()
+            return list(map(lambda x: x.to_mlflow_entity(), users))
```

Comment:
```suggestion
            return [u.to_mlflow_entity() for u in users)]
```

list comprehension is simpler